src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetBuildManifestMetadataTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private string testManifestFolderPath = @".\TestManifests\"; 
+
+        public GetBuildManifestMetadataTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+        [Test]
+        public void EmptyManifestFolderPath()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata("", new CancellationToken());
+            act.Should().Throw<ArgumentException>().WithMessage("The path is empty. (Parameter 'path')");
+        }
+
+        [Test]
+        public void ParseBasicManifest()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "OneManifest", new CancellationToken());
+            data.Item1.Should().BeEquivalentTo(SharedObjects.ExpectedManifestMetadata);
+            data.Item2.Should().BeEquivalentTo(SharedObjects.ExpectedSigningInfo);
+            SharedObjects.CompareManifestBuildData(data.Item3, SharedObjects.ExpectedManifestBuildData);
```

Comment:
See my other comment regarding SharedObjects. Looking at this test as it is, I'd have to go to SharedObjects to see what our expected value should be instead of having it in the test.

Reply:
Objects moved.